Error in user_cache_dir when XDG_CACHE_HOME is set

If XDG_CACHE_HOME is set, the function `user_cache_dir` fails in

```
>               path = os.path.join(path, appname)
E               UnboundLocalError: local variable 'path' referenced before assignment
```

--- a/numina/tests/test_testcache.py
+++ b/numina/tests/test_testcache.py
@@ -18,13 +18,10 @@
 #
 
 import os
-
 import pytest
 
 from .testcache import user_cache_dir
 
-
-@pytest.mark.xfail(reason='issue #98')
 def test_user_cache_dir_linux_home(monkeypatch, tmpdir):
 
     home = tmpdir.mkdir('hometest')
@@ -34,16 +31,19 @@ def test_user_cache_dir_linux_home(monkeypatch, tmpdir):
     monkeypatch.setattr(os, 'environ', modenviron)
 
     cachedir = os.path.join(home.strpath, '.cache')
-    expected = os.path.join(cachedir, 'numina', 'astropy')
+    expected = os.path.join(cachedir, 'numina')
+
+    if not os.path.exists(expected):
+        os.makedirs(expected)
 
     assert user_cache_dir("numina") == expected
+    assert os.path.exists(os.path.join(expected,'astropy'))
 
 
-@pytest.mark.xfail(reason="issue #99")
 def test_user_cache_dir_linux_xdg(monkeypatch, tmpdir):
 
     home = tmpdir.mkdir('hometest')
-    cache = tmpdir.mkdir('cache')
+    cache = tmpdir.mkdir('.cache')
 
     modenviron = {'HOME': home.strpath,
                   'XDG_CACHE_HOME': cache.strpath
@@ -52,6 +52,7 @@ def test_user_cache_dir_linux_xdg(monkeypatch, tmpdir):
     monkeypatch.setattr(os, 'environ', modenviron)
 
     cachedir = cache.strpath
-    expected = os.path.join(cachedir, 'numina', 'astropy')
+    expected = os.path.join(cachedir, 'numina')
 
     assert user_cache_dir("numina") == expected
+    assert os.path.exists(os.path.join(expected,'astropy'))

--- a/numina/tests/testcache.py
+++ b/numina/tests/testcache.py
@@ -48,10 +48,12 @@ def user_cache_dir(appname=None):
         else:
             if appname in os.environ['XDG_CACHE_HOME']:
                 path = os.environ['XDG_CACHE_HOME'].split(appname)[0]
+            else:
+                path = os.getenv('XDG_CACHE_HOME', os.path.expanduser('~/.cache'))
         if appname:
             path = os.path.join(path, appname)
     if not os.path.exists(os.path.join(path, 'astropy')):
-        os.mkdir(os.path.join(path, 'astropy'))
+        os.makedirs(os.path.join(path, 'astropy'))
     return path
 
 

